src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -175,7 +176,20 @@ public void rename() {
                     Localization.lang("Cancel"));
 
             if (confirm) {
-                pdfCleanup.cleanup(entry);
+                Optional<Path> fileConflictCheck = pdfCleanup.fileAlreadyExists(linkedFile, entry);
+                if (!fileConflictCheck.isPresent()) {
+                    pdfCleanup.cleanup(entry);
+                } else {
+                    confirm = dialogService.showConfirmationDialogAndWait(
```

Comment:
You could reduce code-duplication by extracting this method call into a private method.